addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/Utils.java

```diff
@@ -167,4 +167,18 @@ public static String createJsonString(Map<String, String> params) {
         json.accumulateAll(params);
         return json.toString();
     }
+
+    public static String addCacheBusterParameter(String url, String cacheBuster, String value) {
+        StringBuilder newUrl = new StringBuilder(url);
+
+        if (url.contains("?")) {
+            newUrl.append('&');
+        } else {
+            newUrl.append('?');
+        }
+        newUrl.append(cacheBuster);
+        newUrl.append('=');
+        newUrl.append(value);
+        return newUrl.toString();
+    }
```

Comment:
For future consideration, can this not use ZAPs interal methods for dealing with params instead of string parsing?

Reply:
I guess yes.. which method tho?

Reply:
commonlibs?

Reply:
No there's stuff in core, on HttpMessage/HttpRequest and related.

Reply:
Snippet from elsewhere:
```java
TreeSet<HtmlParameter> urlParams = msg.getUrlParams();
			for (HtmlParameter param : urlParams) {
				if (param.getName().equals(paramName)) {
					param.setValue(attackVector);
				}
			}
			msg.setGetParams(urlParams); // setParameter and setEscapedParameter results in spaces being + vs %20 (or actual space)
```

Reply:
why is it so complicated!? I mean :) a complete new Object? :)

Reply:
Seems more simple than the string parsing to me. Especially given that ?=& are only assumed characters, the RFC actually allows others. But like I said it's an item for the future, we're not going to block this PR on it.

I also wasn't suggesting that's the exact code needed here, just an example of the methods that might be relevant.

Reply:
I believe, all you'd actually need to do is get the set of params, add the cache buster param/value, and set the set to the message again.